packages/client/hmi-client/src/components/projects/NewProjectCard.vue

```diff
@@ -1,16 +1,49 @@
 <script setup lang="ts">
 import Button from '@/components/Button.vue';
 import IconAddFilled32 from '@carbon/icons-vue/es/add--filled/32';
+import Modal from '@/components/Modal.vue';
+import { ref } from 'vue';
+import { useRouter } from 'vue-router';
 
-function createNewProject() {}
+const isModalVisible = ref(false);
+const newProjectTitle = ref('New Project');
+const router = useRouter();
+
+function postProject(name: string) {
+	// Placeholder implementation - replace with api call
+	return {
+		id: '3',
+		name,
+		description: 'This is a new project'
+	};
+}
+
+function createNewProject() {
+	isModalVisible.value = false;
+	const { id } = postProject(newProjectTitle.value);
+	const routeLocation = router.resolve({ path: `/projects/${id}` });
```

Comment:
I thought it would be good to use the RoutePath enum here, but that lead to a circular import.

Reply:
This will be solved by @mwdchang PR